runner/test/cases/server/config.test.js

```diff
@@ -40,6 +42,18 @@ suite(`Server config validation`, () => {
     );
   });
 
+  test("it succeeds when PAY_RETURN_URL is insecure and the environment is test", () => {
+    const configWithInsecureUrl = {
+      payReturnUrl: "http://insecure.url",
+      env: "test",
+      apiEnv: "test",
```

Comment:
I think also if env is production, apiEnv will override the behaviour which is fine but worth noting. Worth writing a test for it or adding documentation for it in default.js.